Source/WebCore/loader/ResourceMonitor.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "ResourceMonitor.h"
+
+#include "LocalFrame.h"
+#include <wtf/CryptographicallyRandomNumber.h>
+
+namespace WebCore {
+
+#if ENABLE(CONTENT_EXTENSIONS)
+
+static constexpr size_t sNetworkUsageThreshold = 4 * 1024 * 1024; // 4MB
+static constexpr size_t sNetworkUsageThresholdRandomness = 1303 * 1024; // 1303KB
+
+static size_t networkUsageThresholdWithRandomNoise()
+{
+    return sNetworkUsageThreshold + static_cast<size_t>(cryptographicallyRandomUnitInterval() * sNetworkUsageThresholdRandomness);
+}
+
+ResourceMonitor::ResourceMonitor(LocalFrame& frame)
+    : m_frame(frame)
+    , m_networkUsageThreshold { networkUsageThresholdWithRandomNoise() }
```

Comment:
What is this used for? I don't see it used in this PR. We should not land dead code.